src/Lykke.Job.HistoryExportBuilder.AzureRepositories/BlobUploader.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using AzureStorage;
+using Lykke.Job.HistoryExportBuilder.Core.Domain;
+using Lykke.Job.HistoryExportBuilder.Core.Services;
+
+namespace Lykke.Job.HistoryExportBuilder.AzureRepositories
+{
+    public class BlobUploader : IFileUploader
+    {
+        private readonly IBlobStorage _blobStorage;
+        private const string ContainerName = "historyexports";
+
+        public BlobUploader(IBlobStorage blobStorage)
+        {
+            _blobStorage = blobStorage;
+        }
+        
+        public async Task<Uri> UploadAsync(string id, FileType type, MemoryStream file)
+        {
+            var key = $"{id}.{GetExtention(type)}";
+
+            if (await _blobStorage.HasBlobAsync(ContainerName, key))
+                return new Uri(await Task.Run(() => _blobStorage.GetBlobUrl(ContainerName, key)));
```

Comment:
why do u need to use Task.Run here?

Reply:
because there is no async version of _blobStorage.GetBlobUrl

Reply:
Why do u need async version?
Why not just use it like this? - new Uri(_blobStorage.GetBlobUrl(ContainerName, key));